### PR TITLE
Rework SQL Server tests to use BaseConnectorTest

### PR DESCRIPTION
**Pull request by alexjo2144**
Refactor SQL Server tests to use the single connector base test class added in https://github.com/trinodb/trino/pull/6989

### CODE REVIEW

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
@alexjo2144 see build red

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java [hunk 1]
```diff
@@ -56,11 +53,71 @@ protected QueryRunner createQueryRunner()
...
+            // TODO this should either work or fail cleanly
+            return Optional.empty();
+        }
+
+        return Optional.of(dataMappingTestSetup);
     }
 
     @Test
-    public void testInsert()
+    public void testSimpleInsert()
```
Comment (findepi) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java:120]: Can this test be considered superseded by `testInsert` from the superclass?

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java [hunk 2]
```diff
@@ -80,7 +137,7 @@ public void testInsertInPresenceOfNotSupportedColumn()
     }
 
     @Test
-    public void testView()
+    public void testSimpleView()
```
Comment (findepi) [inline plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java:140]: Can this test be considered superseded by `testView` from the superclass?

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
see red

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java (hunk 1 above)
Comment (alexjo2144) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java]: This one can be, I'll remove it.

File: plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java (hunk 2 above)
Comment (alexjo2144) [reply plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestSqlServerConnectorTest.java]: Not this one, this creates a view in SqlServer but testView in the superclass creates it in Trino.

**Review by findepi — APPROVED** [review]

**Pull request merged**